community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/NodeCorrectlyIndexedCheck.java

```diff
@@ -114,6 +117,24 @@ private void verifyNodeCorrectlyIndexedUniquely(
         }
     }
 
+    private PrimitiveLongIterator filterIfMultipleValuesFound( int propertyKeyId, Object propertyValue,
+            PrimitiveLongIterator indexedNodeIds )
+    {
+        if ( ((PrimitiveLongPeekingIterator) indexedNodeIds).hasMultipleValues() )
```

Comment:
Why is there a difference if there are multiple values?

It seems to me like it would be just as possible for the index to return an inexact match even if it only returns one match.